File: tests/cases/exceptions.py
#!/usr/bin/env python3


def show():
    s = []
    try:
        raise Exception("foo")
    except Exception as e:
        s.append("foo")
    finally:
        s.append("Finally")

    try:
        3 / 0
    except ZeroDivisionError:
        s.append("ZeroDivisionError")

    try:
        raise Exception("foo")
    except:
        s.append("foo_2")

    return s

    try:
        raise Exception("foo")
    except Exception as e:
        assert "foo" in str(e)


if __name__ == "__main__":
    assert show() == ["foo", "Finally", "ZeroDivisionError", "foo_2"]

File: tests/cases-py2many/exceptions.jl
function show()::Vector
    s = []
    try
        throw(Exception("foo"))
    catch exn
        let e = exn
            if e isa Exception
                push!(s, "foo")
            end
        end
    finally
        push!(s, "Finally")
    end
    try
        3 / 0
    catch exn
        if exn isa ZeroDivisionError
            push!(s, "ZeroDivisionError")
        end
    end
    try
        throw(Exception("foo"))
    catch exn
        push!(s, "foo_2")
    end
    return s
end

function main()
    @assert(show() == ["foo", "Finally", "ZeroDivisionError", "foo_2"])
end

main()
